### PR TITLE
Removed already deleted backups to prevent backup delete from failing

### PR DESCRIPTION
**Pull request by snigdha-px**
**What this PR does / why we need it**:
1)Removed already deleted backups to prevent backup delete from failing
2)Added verifysafely instead of verifyfatal to prevent cleaning up in case some cleanup in between fails
**Which issue(s) this PR fixes** (optional)
Closes #
Aetso : http://aetos.pwx.purestorage.com/resultSet/testSetID/89067
jenkins: https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test/85/console
**Special notes for your reviewer**:

### CODE REVIEW

File: drivers/backup/auth.go [hunk 1]
```diff
@@ -1045,7 +1048,8 @@ func AddGroupToUser(user, group string) error {
 		log.Errorf("%s: %v", fn, err)
 		return err
 	}
-	_, err = processHTTPRequest(method, reqURL, headers, strings.NewReader(string(dataBytes)))
+	response, err := processHTTPRequest(method, reqURL, headers, strings.NewReader(string(dataBytes)))
```
Comment (lsrinivas-pure) [inline drivers/backup/auth.go:1051]: how are you handling failed responses?

File: tests/backup/backup_test.go [hunk 2]
```diff
@@ -2933,16 +2931,22 @@ var _ = Describe("{ShareBackupWithDifferentRoleUsers}", func() {
 			go func(backupName string) {
 				defer wg.Done()
 				backupUID, err := backupDriver.GetBackupUID(ctx, backupName, orgID)
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:2933]: add ginkgorecover here

File: tests/backup/backup_test.go [hunk 3]
```diff
@@ -5389,21 +5401,27 @@ var _ = Describe("{ShareAndRemoveBackupLocation}", func() {
...
 		ValidateAndDestroy(contexts, opts)
 		backupDriver := Inst().Backup
-		for _, backupName := range backupNames {
+		for _, backupName := range newBackupNames {
 			wg.Add(1)
 			go func(backupName string) {
 				defer wg.Done()
 				backupUID, err := backupDriver.GetBackupUID(ctx, backupName, orgID)
-				log.FailOnError(err, "Error deleting user %v", backupName)
+				dash.VerifySafely(err, nil, fmt.Sprintf("Getting backup UID for backup %v", backupName))
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:5409]: same here

**Review by lsrinivas-pure — CHANGES_REQUESTED** [review]

File: tests/backup/backup_test.go (hunk 3 above)
Comment (snigdha-px) [reply tests/backup/backup_test.go:5411]: done!!

File: tests/backup/backup_test.go (hunk 2 above)
Comment (snigdha-px) [reply tests/backup/backup_test.go:2934]: done

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Pull request merged**